src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/WaySectionProcessor.java

```diff
@@ -0,0 +1,906 @@
+        {
+            logger.error("Line {} hass less than {} points, cannot be sectioned!",
+                    line.getIdentifier(), MINIMUM_POINTS_TO_QUALIFY_AS_A_LINE);
+            return newEdgesForLine;
+        }
+
+        // Prepare the nodes identifiers, identifier factory and one way information
+        final NodeOccurrenceCounter nodesToSectionAt = changeSet.getNodesForEdge(line);
+        final WaySectionIdentifierFactory identifierFactory = new WaySectionIdentifierFactory(
+                line.getIdentifier());
+
+        // Determines if we need to reverse the polyline and if a reverse edge is needed
+        final PbfOneWay oneWay = PbfOneWay.forTag(line);
+        final boolean hasReverseEdge = oneWay == PbfOneWay.NO;
+        final boolean isReversed = oneWay == PbfOneWay.REVERSED;
+
+        // Keep track of the start index, start/end nodes
+        int startIndex;
+        Optional<TemporaryNode> startNode = Optional.empty();
+        Optional<TemporaryNode> endNode = Optional.empty();
+
+        try
+        {
+            // Find the first node
+            startIndex = 0;
+            startNode = nodesToSectionAt.getNode(polyline.first());
+            if (!startNode.isPresent())
+            {
+                // Weren't able to find the starting node, abort slicing
+                logger.error(
+                        "Could not find starting Node for Line {} during way-sectioning. Aborting!",
+                        line.getIdentifier());
+                return newEdgesForLine;
+            }
+
+            // We've already processed the starting node, so we start with the first index
+            for (int index = 1; index < polyline.size(); index++)
+            {
+                // Check to see if this location is a node
+                endNode = nodesToSectionAt.getNode(polyline.get(index));
+                if (endNode.isPresent())
+                {
+                    // We found the end node, create the edge. Note: using occurrence minus one
+                    // since PolyLine uses zero-based numbering. We are incrementing only the
+                    // start node occurrence, since the end node will either be used as a future
+                    // start node or be the end of the way, in which case we don't care.
+                    final int startOccurrence = nodesToSectionAt.getOccurrence(startNode.get()) - 1;
+                    nodesToSectionAt.incrementOccurrence(startNode.get());
+                    final int endOccurrence = nodesToSectionAt.getOccurrence(endNode.get()) - 1;
+
+                    // Build the underlying polyline and reverse it, if necessary
+                    final PolyLine rawPolyline = polyline.between(polyline.get(startIndex),
+                            startOccurrence, polyline.get(index), endOccurrence);
+                    final PolyLine edgePolyline = isReversed ? rawPolyline.reversed() : rawPolyline;
```

Comment:
in the case of a reversed feature, would the resulting identifiers be reversed - and is it expected?  (poorly phrased so an example below).  NOTE: I maybe have confused myself on this, so let me know if I made any sense here.

Let's say line A->B->C is a reversed line to be sectioned at every point.
1. rawPolyline 1 = A->B, and it will be given identifier 1 and the geometry will be B->A
2. rawPolyline 2 = B->C, and it will be given identifier 2 and the geometry will be C->B.

the resulting edge will have section 001 being B->A and then 002 being C->B.

Reply:
The identifiers would not be reversed, they would occur in the order of the shape point occurrence. Only the geometry is affected during the reverse. To summarize, your example above is exactly what would happen :) 

This is expected. I compared it with today's code and atlas files for this way (https://www.openstreetmap.org/way/333112568#map=19/5.31774/-4.01493) and there is exact parity.

Before:
<img width="1049" alt="screen shot 2018-02-21 at 12 44 38 pm" src="https://user-images.githubusercontent.com/18317260/36504470-77ce1134-1705-11e8-8d9c-ab9dbff84c27.png">

After: 
<img width="1060" alt="screen shot 2018-02-21 at 12 44 48 pm" src="https://user-images.githubusercontent.com/18317260/36504492-86e9ccb2-1705-11e8-9648-9f0a12ddf203.png">

Reply:
nice! Thanks for the detailed pictures as well!